miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/MiniAppCustomPermissionCache.kt

```diff
@@ -96,44 +96,17 @@ internal class MiniAppCustomPermissionCache(context: Context) {
         applyStoringPermissions(MiniAppCustomPermission(miniAppId, allPermissions))
     }
 
-    @SuppressWarnings("PrintStackTrace")
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun applyStoringPermissions(miniAppCustomPermission: MiniAppCustomPermission) {
-        try {
-            val jsonToStore: String = Gson().toJson(orderByDefaultList(miniAppCustomPermission))
-            prefs.edit().putString(miniAppCustomPermission.miniAppId, jsonToStore).apply()
-        } catch (e: Exception) {
-            e.printStackTrace()
-        }
+        val jsonToStore: String = Gson().toJson(sortedByDefault(miniAppCustomPermission))
+        prefs.edit().putString(miniAppCustomPermission.miniAppId, jsonToStore).apply()
     }
 
-    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
-    fun orderByDefaultList(miniAppCustomPermission: MiniAppCustomPermission): MiniAppCustomPermission {
-        val miniAppId = miniAppCustomPermission.miniAppId
-        val defaultTypesOrder = mutableListOf<MiniAppCustomPermissionType>()
-        defaultDeniedList(miniAppId).pairValues.forEach {
-            defaultTypesOrder.add(it.first)
-        }
-
-        val currentTypesOrder = mutableListOf<MiniAppCustomPermissionType>()
-        miniAppCustomPermission.pairValues.forEach {
-            currentTypesOrder.add(it.first)
-        }
-
-        val expectedTypesOrder = currentTypesOrder.map {
-            defaultTypesOrder.indexOf(it)
-        }.sorted().map { value -> defaultTypesOrder[value] }
-
-        val orderedPair =
-            mutableListOf<Pair<MiniAppCustomPermissionType, MiniAppCustomPermissionResult>>()
-
-        expectedTypesOrder.forEachIndexed { index, type ->
-            miniAppCustomPermission.pairValues.find {
-                it.first == type
-            }?.let { orderedPair.add(index, it) }
-        }
-
-        return MiniAppCustomPermission(miniAppId, orderedPair)
+    // Sort the `pairValues` by ordinal of [MiniAppCustomPermissionType].
+    @VisibleForTesting
+    internal fun sortedByDefault(miniAppCustomPermission: MiniAppCustomPermission): MiniAppCustomPermission {
+        val sortedPairValues = miniAppCustomPermission.pairValues.sortedBy { it.first.ordinal }
```

Comment:
In previous implementation, I wanted to use the ordering from `defaultDeniedList`, the expected order would be inside this class. But, using `MiniAppCustomPermissionType` is better as it removes some boilerplate code 👍 As well as the related unit test has been passed.

P/S: I don't think, we need to use `internal` here, this entire class is already `internal`.